### PR TITLE
Simplify Gson poms.

### PR DESCRIPTION
**Pull request by swankjesse**
A lot of the configuration in these poms has been taken over by improvements
to the Maven release plugin.

### CODE REVIEW

**Comment by JakeWharton** [comment]
LGTM. I would do a `mvn clean javadoc:jar sources:jar` to verify those tasks do what you expect and generate the output we expect.

File: gson/pom.xml [hunk 1]
```diff
@@ -1,246 +1,36 @@
...
-            <goals>
-              <goal>jar</goal>
-            </goals>
-          </execution>
-        </executions>
-        <configuration>
-          <archive>
-            <addMavenDescriptor>false</addMavenDescriptor>
-            <manifestEntries>
-              <Bundle-RequiredExecutionEnvironment>J2SE-1.5, JavaSE-1.6, JavaSE-1.7, JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
```
Comment (inder123) [inline gson/pom.xml]: This was added specifically for OSGI. How will that requirement be met?
Disclaimer: I don't fully understand what the requirement was.

File: pom.xml [hunk 2]
```diff
@@ -1,249 +1,97 @@
...
   <url>https://github.com/google/gson</url>
-  <description>Google Gson library</description>
+
+  <modules>
+    <module>gson</module>
+  </modules>
+
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <java.version>1.7</java.version>
```
Comment (inder123) [inline pom.xml]: Why 1.7? I thought we went up to 1.6?

Comment (swankjesse) [inline pom.xml]: Oh wow, that’s a big bug. Will fix.

File: gson/pom.xml (hunk 1 above)
Comment (swankjesse) [inline gson/pom.xml]: Yeah, I don’t think this is the current way that OSGi works anyways. (There’s a special bundle target now).

File: pom.xml (hunk 2 above)
Comment (swankjesse) [inline pom.xml]: https://github.com/google/gson/pull/789

**Pull request merged**